### PR TITLE
fix 'any endpoints' for axis style inputs

### PR DESCRIPTION
**Pull request by ZappoMan**

### CODE REVIEW

**Comment by ZappoMan** [comment]
This fixes this bug: https://highfidelity.fogbugz.com/f/cases/1679/Vive-controller-trackpad-input-only-reports-axis-values-from-0-1-rather-than-1-1

The bug was that if a script attempted to set up an "any" filter, that filtered multiple inputs and those inputs were axis that ranged from -1 to 1, then the negative values would be clamped. This problem did not exist for scripts that didn't use an array/any filter.

Test Plan:
- smoke test, make sure input controllers like Vive, XBox, and/or Touch still work.
- Run this script in the Console:
```
var mapping = Controller.newMapping("viveTesting");
mapping.from([Controller.Hardware.Vive.LX]).peek().to(function(value) { print("Vive.LX: ", value) });
mapping.from([Controller.Hardware.Vive.RY]).peek().to(function(value) { print("Vive.RY: ", value) });
Controller.enableMapping("viveTesting");
```

As you use the vive controller and move your finger no the thumbpad, you should see values from -1 to 1.

**Comment by hifi-gustavo** [comment]
Build console output for commit: a87c06c3c8c9014d5f92f1ed4eb26ec45a50d7a0 available at: https://gist.github.com/8f992a9639e54641a480990f73c69d9a<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8767-a87c06c.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8767-a87c06c.dmg

File: libraries/controllers/src/controllers/impl/endpoints/AnyEndpoint.cpp [hunk 1]
```diff
@@ -27,10 +27,15 @@ AnyEndpoint::AnyEndpoint(Endpoint::List children) : Endpoint(Input::INVALID_INPU
     }
 }
 
+// The value of an any-point is considered to be the maxiumum absolute value,
+// this handles any's of multiple axis values as well as single values as well
 float AnyEndpoint::peek() const {
```
Comment (jherico) [inline libraries/controllers/src/controllers/impl/endpoints/AnyEndpoint.cpp]: you need to add the same logic in `AnyEndpoint::value()`

Comment (ZappoMan) [inline libraries/controllers/src/controllers/impl/endpoints/AnyEndpoint.cpp]: fixed

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8da9fccb5d7f4947f4527568ae6b37f5e4be9ac7 available at: https://gist.github.com/8f992a9639e54641a480990f73c69d9a<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8767-8da9fcc.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8767-8da9fcc.dmg

**Comment by kevinmthomasse** [comment]
testing

**Comment by kevinmthomasse** [comment]
## QA Test
**QA Tester**: Unknown user
**localhost 'createTestBlocks.js' Result**: **localhost 'createTestBlocks.js'** SUCCESS
**localhost ATP Upload + Edit Result**: **localhost ATP** SUCCESS
**Echo Server Audio**: **Echo Server Audio** SUCCESS
**localhost 'harmonicOscillator.js' Result**: **localhost 'harmonicOscillator.js'** SUCCESS
**localhost 'reverbTest.js' Result**: **localhost 'reverbTest.js'** SUCCESS
**localhost 'playaPerformanceTest.js' Result**: **localhost 'playaPerformanceTest.js'** SUCCESS
**dev-playa 'playaPerformanceTest.js' Result**: **dev-playa 'playaPerformanceTest.js'** SUCCESS
**dev-demo 'playaPerformanceTest.js' Result**: **dev-demo 'playaPerformanceTest.js'** SUCCESS
**QA Pass**: **QA Pass** SUCCESS
**Additional Test Results + Observations**:  Only have Hydras and tested with them in ```dev-demo``` no issue of any kind.  Script obviously didn't work as I believe it is meant for Vive as I received a reference error because I was using Hydras.

**Comment by thoys** [comment]
Made a quick testrun with the script in console, all works fine for me with the Vive controllers.  On the left controller the left to right values don't exceed the range of -1 to 1 and for the  right controller the bottom to top values don't exceed the range of -1 to 1.

💯

**Pull request merged**